src/Databases/MySQL/MaterializeMetadata.cpp

```diff
@@ -239,9 +237,16 @@ MaterializeMetadata::MaterializeMetadata(
         assertString("\nData Version:\t", in);
         readIntText(data_version, in);
 
-        if (checkBinlogFileExists(connection))
-            return;
     }
+}
+
+void MaterializeMetadata::startReplication(
```

Comment:
Would it be a good idea to maintain a "running" state in MaterializeMetadata, and add a "stopReplication" method?

Reply:
Perhaps.  It is not necessary now though, so I think that should be added when the need arises.